plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts

```diff
@@ -109,7 +114,7 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
     // set a nonce cookie before redirecting to oauth provider
     this.setNonceCookie(res, nonce, cookieConfig);
 
-    const state: OAuthState = { nonce, env, origin };
+    const state: OAuthState = { nonce, env, origin, redirectUrl, flow: flow };
```

Comment:
```suggestion
    const state: OAuthState = { nonce, env, origin, redirectUrl, flow };
```

While we're at it :grin: